lms/static/scripts/frontend_apps/utils/google-picker-client.js

```diff
@@ -63,54 +66,94 @@ export class GooglePickerClient {
     this._developerKey = developerKey;
     this._origin = addHttps(origin);
 
-    const libs = loadLibraries(['auth2', 'client', 'picker']);
+    const libs = loadLibraries(['client', 'picker']);
+
+    this._identityServices = loadIdentityServicesLibrary();
 
-    this._gapiAuth2 = libs.then(({ auth2 }) => auth2);
     this._gapiClient = libs.then(({ client }) => client);
     this._gapiPicker = libs.then(({ picker }) => picker.api);
+
+    /** @type {string|null} */
+    this._accessToken = null;
   }
 
   /**
    * Authorize this application's access to the user's files in Google Drive.
    *
    * @return {Promise<string>} - An access token for making Google Drive API requests.
+   *   The access token is cached for use in future Drive API calls.
    */
-  async _authorizeDriveAccess() {
-    const auth2 = await this._gapiAuth2;
-    const googleAuth = auth2.init({
+  async requestAuthorization() {
```

Comment:
Previously the only way to authorize Google Drive API access was implicitly by calling  `GooglePickerClient.showPicker` first. That still works, but the re-use of the access token is now explicit within the class, and users of `GooglePickerClient` can also call `requestAuthorization` directly, as the tests for `enablePublicViewing` do.

Reply:
Is there any case where there could be a need to refresh that access token?